File: src/main/java/cn/booklish/sharp/util/GsonUtil.java
package cn.booklish.sharp.util;

import com.google.gson.Gson;

/**
 * @author: 刘新冬(www.booklish.cn)
 * @data: 2017/12/4 21:03
 * @desc:
 */
public class GsonUtil {

    private static final Gson gson = new Gson();

    public static String toJson(Object source){
        return gson.toJson(source);
    }

    public static <T> T jsonToObject(String json,Class<T> clazz){
        return gson.fromJson(json,clazz);
    }

}

File: src/main/java/cn/booklish/sharp/util/GsonUtil.kt
package cn.booklish.sharp.util

import com.google.gson.Gson

/**
 * @Author: liuxindong
 * @Description:  json转化工具
 * @Created: 2017/12/13 9:05
 * @Modified:
 */
class GsonUtil {

    companion object{

        private val gson:Gson = Gson()

        fun objectToJson(obj: Any):String{
            return gson.toJson(obj)
        }

        fun <T> jsonToObject(json:String,clazz:Class<T>):T{
            return gson.fromJson(json,clazz)
        }
    }

}